guppylang/prelude/builtins.py

```diff
@@ -882,8 +882,45 @@ def print(x): ...
 def property(x): ...
 
 
-@guppy.custom(builtins, checker=UnsupportedChecker(), higher_order_value=False)
-def range(x): ...
+@guppy.struct(builtins)
+class Range:
+    stop: int
+
+    @guppy(builtins)
+    def __iter__(self: "Range") -> "RangeIter":
+        return RangeIter(0, self.stop)  # type: ignore[call-arg]
```

Comment:
Still lots of `type: ignore` here even with the `@overload`s in `decorator.py`. Maybe we should just disable mypy for the whole file??

Reply:
There is a `mypy: ignore` pragma at the start of the file. You can add `call-arg` to that.

Alternatively, you can add a `@no_type_check` decorator to the function

Reply:
`@no_type_check` does work, but still has to be done in three places.

I've gone with whole-file `mypy: disable-error-code`, but the killer here is that this makes the `disable-error-code` line too long (so `ruff` complains), and I couldn't see any way to disable the line-length check for just that line (if you stick a `# noqa: E501` on the end, ruff doesn't recognize this because it's within an existing comment, and mypy takes that as part of an error-code to disable, doesn't recognize it and barfs everywhere...)

So, significant cost to disabling `call-arg` across the file - perhaps best to run with three separate `#type: ignore`s?

Reply:
> Oof that's annoying, I presume you can't do multiple `mypy: disable-error-code` lines?

Sadly not - see e.g. https://github.com/python/mypy/issues/14750

> I'm happy with `@no_type_check` or `#type: ignore`

Ok, I'll reinstate the type-ignores.